gh-106535: Soft deprecate the getopt module (#105735)
python/cpython: Doc/library/getopt.rst

**Source code:** :source:`Lib/getopt.py`

.. deprecated:: 3.13
   The :mod:`getopt` module is deprecated and will not be developed further;

>    development will continue with the :mod:`argparse` module.

Reviewer: We should tailor this message (copied from the `optparse` deprecation) to be a little more appropriate here:

* Active development has long since stopped here and continued with `argparse`, as the last `getopt`-specific change (that wasn't cross-stdlib cleanup) was bpo-11621 / gh-55830 in 2011.
* It seems worth mentioning explicitly there are no plans for removal in the near future, as the lack of removal version could created user uncertainty about whether and how urgently users should worry about their code breaking.
* Provide clearer guidance to users about what it is recommended that they do

```suggestion
   The :mod:`getopt` module is deprecated, as the actively-developed
   :mod:`argparse` module offers much more functionality with less user code.
   While there are no immediate removal plans, migrating to :mod:`!argparse`
   is recommended.
```

.. note::

   The :mod:`getopt` module is a parser for command line options whose API is

>    designed to be familiar to users of the C :c:func:`!getopt` function. Users who
>    are unfamiliar with the C :c:func:`!getopt` function or who would like to write
>    less code and get better help and error messages should consider using the
>    :mod:`argparse` module instead.

Reviewer: ```suggestion
   designed to be familiar to users of the C :c:func:`!getopt` function.
   Using the :mod:`argparse` module is encouraged instead,
   as it offers better help and error messages while writing less code.
```

Tweaked this a bit to account to nudge users more toward `argparse` rather than just naming specific categories of users that should "consider using" it instead, as well as be more concise and less duplicative/repetitive.

Reviewer: Note that I still think these changes (or perhaps a revised version) would be a good idea if we decide not to actually deprecate it, to make this clearer, more concise and a bit more of a nudge.
